mmv1/third_party/terraform/services/monitoring/resource_monitoring_dashboard.go

```diff
@@ -14,24 +14,45 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
 )
 
-func monitoringDashboardDiffSuppress(k, old, new string, d *schema.ResourceData) bool {
-	computedFields := []string{"etag", "name"}
+// This recursive function takes an old map and a new map and is intended to remove the computed keys
+// from the old json string (stored in state) so that it doesn't show a diff if it's not defined in the
+// new map's json string (defined in config)
+func removeComputedKeys(old map[string]interface{}, new map[string]interface{}) map[string]interface{} {
+	for k, v := range old {
+		if _, ok := old[k]; ok && new[k] == nil {
+			delete(old, k)
+			continue
+		}
+
+		if reflect.ValueOf(v).Kind() == reflect.Map {
+			old[k] = removeComputedKeys(v.(map[string]interface{}), new[k].(map[string]interface{}))
+			continue
+		}
+
+		if reflect.ValueOf(v).Kind() == reflect.Slice {
+			for i, j := range v.([]interface{}) {
+				if reflect.ValueOf(j).Kind() == reflect.Map {
```

Comment:
Do we also need to check for slices inside slices?

I *think* something like this could be valid:
{ "data": [ [ { "something" } ] ] }

Reply:
I'll test this out

Reply:
@slevenick I couldn't get a working configuration that used nested slices like this, let alone ones that would have computed values. Also, looking through all of the [examples here](https://cloud.google.com/monitoring/dashboards/api-examples) an instance of [[  ]] does not appear.

It doesn't seem worthwhile add for the edge case if I can't test it in config?